sdk/java/pulumi/src/test/java/com/pulumi/serialization/internal/ResourceRefPropertyTest.java

```diff
@@ -30,13 +30,22 @@
 import java.util.Optional;
 import java.util.concurrent.CompletableFuture;
 
+import static com.pulumi.core.OutputTests.waitFor;
+import static com.pulumi.core.OutputTests.waitForValue;
 import static com.pulumi.deployment.internal.DeploymentTests.cleanupDeploymentMocks;
 import static com.pulumi.serialization.internal.ConverterTests.deserializeFromValue;
 import static com.pulumi.serialization.internal.ConverterTests.serializeToValueAsync;
 import static org.assertj.core.api.Assertions.assertThat;
 
 class ResourceRefPropertyTest {
 
+    private static final Class<ImmutableMap<String, String>> ValuesClass =
+            TypeShape.<ImmutableMap<String, String>>builder(ImmutableMap.class)
```

Comment:
TypeShape, really :) Interesting. Java can't do it with STDLIB alone?

Reply:
I've checked it does not compile.